src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpRequestNode.cs

```diff
@@ -85,11 +88,57 @@ public HttpBindingResult<HttpRequestMessage> TryGetHttpRequestMessage(HttpBindin
         }
         else
         {
-            success = false;
             diagnostics.AddRange(uriBindingResult.Diagnostics);
         }
 
-        if (success)
+        var headers =
+            HeadersNode?.HeaderNodes.Select(h => new KeyValuePair<string, string>(h.NameNode.Text, h.ValueNode.Text)).ToArray()
```

Comment:
The headers could also contain expressions. But I guess we will flesh that out in future PRs cc @bleaphar